src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -67,9 +68,81 @@ public static partial class Interlocked
         /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static short Exchange(ref short location1, short value) =>
+            (short)Exchange(ref Unsafe.As<short, ushort>(ref location1), (ushort)value);
+
+        /// <summary>Sets a 8-bit unsigned integer to a specified value and returns the original value, as an atomic operation.</summary>
+        /// <param name="location1">The variable to set to the specified value.</param>
+        /// <param name="value">The value to which the <paramref name="location1"/> parameter is set.</param>
+        /// <returns>The original value of <paramref name="location1"/>.</returns>
+        /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Is AggressiveInlining here just a copy&paste? It does not sound like a good idea to aggressively inline all this.

Reply:
I assume this should be similar in size to what native compilers emit for RISC-V and [they do inline that](https://godbolt.org/z/jT649MexM).

Reply:
One benefit to inlining this is that with #99130 it lets the JIT fold most of the bit operations here when passed a ref to a static field.